### PR TITLE
Add Unit Tests to MovePrimitive Class

### LINKED ISSUE #129: Add Unit Tests For The Move Primitive
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
As best I can tell, there are no tests for the `move_primitive`.

**Describe the solution you'd like**
We should add unit tests that give us coverage over all functions that `move_primitive` implements.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by EvanMorcom**
### Description
This PR adds 3 new getter functions to the MovePrimitive class and unit tests for the entire class.

### Testing Done
Unit tests for all MovePrimitive class functions and Primitive class functions.

### Resolved Issues
resolves #129 

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

**Comment by garethellis0** [comment]
Looking at the code coverage results, looks like there's a few more things we should test, namely constructing a `MovePrimitive` from a ROS message, and the `getExtraBitArray` function.

It also looks like we're not testing anything in `primitive.cpp`, so adding some unit tests there would be good as well.

Here's a link to the coverage results for the `primitive` folder:
https://codecov.io/gh/UBC-Thunderbots/Software/tree/af2c8ee30fbad97f94c36403595ab808019ae438/src/thunderbots/software/ai/primitive

**Review by garethellis0 — DISMISSED** [review]
Looks good to me! :+1: :tada:

File: src/thunderbots/software/ai/primitive/move_primitive.h [hunk 1]
```diff
@@ -26,11 +31,31 @@ class MovePrimitive : public Primitive
      *
      * @param primtiive_msg The message from which to create the Move Primitive
      */
-    explicit MovePrimitive(const thunderbots_msgs::Primitive &primtiive_msg);
+    explicit MovePrimitive(const thunderbots_msgs::Primitive &primtive_msg);
```
Comment (henryymliu) [inline src/thunderbots/software/ai/primitive/move_primitive.h]: primitive_msg

File: src/thunderbots/software/ai/primitive/move_primitive.h [hunk 2]
```diff
@@ -26,11 +31,31 @@ class MovePrimitive : public Primitive
...
+
+    /**
+     * Gets the robots destination orientation
+     *
+     * @return The robots final orientation as an Angle
+     */
+    Angle getFinalAngle() const;
+
+    /**
+     * Gets the robots final speed in m/s
```
Comment (henryymliu) [inline src/thunderbots/software/ai/primitive/move_primitive.h]: nit: robot's

File: src/thunderbots/software/test/primitive/move_primitive.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    EXPECT_EQ(extra_bit_array, std::vector<bool>());
+}
+
+TEST(MovePrimitiveTest, create_primitive_from_message_test)
+{
+    const Point destination  = Point(2, -3);
+    const Angle final_angle  = Angle::ofRadians(3.55);
+    const double final_speed = 2.22;
+    const int robot_id       = 3U;
```
Comment (henryymliu) [inline src/thunderbots/software/test/primitive/move_primitive.cpp]: So you're assigning an unsigned int to an int. Consider declaring it unsigned int at the beginning.

File: src/thunderbots/software/test/primitive/primitive.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+#include "ai/primitive/primitive.h"
+
+#include <gtest/gtest.h>
+
+#include "ai/primitive/move_primitive.h"
+
+TEST(PrimitiveTest, create_message_empty_test)
+{
+    MovePrimitive move_prim = MovePrimitive(int(), Point(), Angle(), double());
```
Comment (henryymliu) [inline src/thunderbots/software/test/primitive/primitive.cpp]: Since you're comparing int() to int() later on, please use an actual declared number.

**Review by henryymliu — CHANGES_REQUESTED** [review]
nits

File: src/thunderbots/software/ai/primitive/move_primitive.h (hunk 1 above)
Comment (EvanMorcom) [reply src/thunderbots/software/ai/primitive/move_primitive.h]: Fixed typo.

File: src/thunderbots/software/ai/primitive/move_primitive.h (hunk 2 above)
Comment (EvanMorcom) [reply src/thunderbots/software/ai/primitive/move_primitive.h]: Fixed

File: src/thunderbots/software/test/primitive/move_primitive.cpp (hunk 3 above)
Comment (EvanMorcom) [reply src/thunderbots/software/test/primitive/move_primitive.cpp]: Set to unsigned.

File: src/thunderbots/software/test/primitive/primitive.cpp (hunk 4 above)
Comment (EvanMorcom) [reply src/thunderbots/software/test/primitive/primitive.cpp]: Removed all int() and double()

**Review by henryymliu — APPROVED** [review]
LGTM

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**